pkg/datapath/iptables/iptables.go

```diff
@@ -334,7 +334,16 @@ func (m *IptablesManager) Init() {
 			"ip6tables kernel modules could not be loaded, so IPv6 cannot be used")
 		ip6tables = false
 	}
-	m.ip6tables = ip6tables
+	m.haveIp6tables = ip6tables
+
+	if err := modulesManager.FindOrLoadModules("xt_socket"); err != nil {
```

Comment:
Shouldn't we probe for the `nf_socket_ipv{4,6}` modules as well?

```
> lsmod | grep xt_socket
xt_socket              16384  0
nf_socket_ipv4         16384  1 xt_socket
nf_socket_ipv6         20480  1 xt_socket
nf_defrag_ipv6         24576  2 nf_conntrack,xt_socket
nf_defrag_ipv4         16384  2 nf_conntrack,xt_socket
x_tables               49152  8 xt_conntrack,iptable_filter,xt_socket,xt_tcpudp,xt_addrtype,xt_nat,ip_tables,xt_MASQUERADE
```

Reply:
Given that they are dependencies, I think that if they are not available then `xt_socket` load will fail anyway. Unless you're concerned about kernels with IPv6 compiled out?

Reply:
Makes sense.